test/sandbox/git.go

```diff
@@ -15,70 +15,87 @@
 package sandbox
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/madlambda/spells/assert"
 	"github.com/mineiros-io/terramate/git"
 	"github.com/mineiros-io/terramate/test"
 )
 
-const baseRemoteName = "origin"
+const defRemote = "origin"
+const defRemoteBranch = "main"
+const defLocalBranch = "main"
 
 // Git is a git wrapper that makes testing easy by handling
 // errors automatically, failing the caller test.
 type Git struct {
-	t       *testing.T
-	g       *git.Git
-	basedir string
+	t          *testing.T
+	g          *git.Git
+	localrepo  string
+	remoterepo string
 }
 
 func NewGit(t *testing.T, basedir string) Git {
 	return Git{
-		t:       t,
-		g:       test.NewGitWrapper(t, basedir, []string{}),
-		basedir: basedir,
+		t:         t,
+		g:         test.NewGitWrapper(t, basedir, []string{}),
+		localrepo: basedir,
 	}
 }
 
-// Init will initialize the git repo with a default local remote.
+// Init will initialize the local git repo with a default remote.
 // After calling Init(), the methods Push() and Pull() pushes and pulls changes
 // from/to the configured default remote.
 func (git Git) Init() {
 	t := git.t
 	t.Helper()
 
-	git.InitBasic()
+	git.InitLocalRepo()
 
 	// the main branch only exists after first commit.
-	path := test.WriteFile(t, git.basedir, "README.md", "# generated by terramate")
+	path := test.WriteFile(t, git.localrepo, "README.md", "# generated by terramate")
 	git.Add(path)
 	git.Commit("first commit")
 
-	git.SetupRemote(baseRemoteName, "main")
+	git.ConfigureDefaultRemote()
 }
 
-// SetupRemote will do basic setup of a remote with the given branch.
-func (git Git) SetupRemote(remote, branch string) {
+func (git Git) ConfigureDefaultRemote() {
+	git.initRemoteRepo(defRemoteBranch)
+	git.RemoteAdd(defRemote, git.remoterepo)
+	// Pushes current branch onto defRemote and defBranch
+	git.PushOn(defRemote, defRemoteBranch, defLocalBranch)
+
+}
+
+// SetupRemote creates a bare remote repository and setup the local repo with it
+// using remoteName and remoteBranch.
+func (git Git) SetupRemote(remoteName, remoteBranch, localBranch string) {
```

Comment:
I could be way off on this, but it seems that SetupRemote should not be called multiple times since it would change git.remoterepo and cause some interesting behavior on the git sandbox, if that is the case I would add a check that if it is calledwith al already setup remote it fails immediately, I had some "interesting" problems by making mistakes on my tests and then the errors that happened were really hard to track down since it fails in some distance place (not the setup) and correlating was hard/time consuming.

If it would work with multople remotes setup then just disregard.

Reply:
Taking a second look it seems that we could setup multiple remotes but then the field `git.remoterepo` is confusing, it seems that it is used as some sort of temporary variable between initialization + remoteAdd and nothing else, but in that case it being a field inside the git instance is misleading/confusing (at least to me), I thought it was integral part of the state of the git instance that was changing everytime, but it actually isn't...I think :thinking:

Reply:
I removed the Git.remoterepo, it wasn't actually needed (at least at this point).

Reply:
yeah if it was fundamental to internal state I would expect that we would not be able to have multiple remotes, or else we would need a slice of remotes instead of a single remote